procedure/src/main/java/org/apache/iotdb/procedure/TimeoutExecutorThread.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.procedure;
+
+import java.util.concurrent.DelayQueue;
+import java.util.concurrent.Delayed;
+import java.util.concurrent.TimeUnit;
+
+public class TimeoutExecutorThread<Env> extends StoppableThread {
+
+  private final ProcedureExecutor<Env> executor;
+
+  private final DelayQueue<ProcedureDelayContainer<Env>> queue = new DelayQueue<>();
+
+  public TimeoutExecutorThread(
+      ProcedureExecutor<Env> envProcedureExecutor, ThreadGroup threadGroup, String name) {
+    super(threadGroup, name);
+    setDaemon(true);
+    this.executor = envProcedureExecutor;
+  }
+
+  public void add(Procedure<Env> procedure) {
+    queue.add(new ProcedureDelayContainer<>(procedure));
+  }
+
+  public boolean remove(Procedure<Env> procedure) {
+    return queue.remove(new ProcedureDelayContainer<>(procedure));
+  }
+
+  private ProcedureDelayContainer<Env> takeQuietly() {
+    try {
+      return queue.poll(20, TimeUnit.SECONDS);
```

Comment:
please create a constant variable for 20

Reply:
fixed